com.unity.render-pipelines.core/Runtime/Debugging/FrameTiming/FrameTimeBottleneck.cs

```diff
@@ -33,6 +33,11 @@ internal struct BottleneckHistogram
     /// </summary>
     internal class BottleneckHistory
     {
+        public BottleneckHistory(int initialCapacity)
```

Comment:
It seems to me that `BottleneckHistory` and `FrameTimeSampleHistory` are pretty the same. I would encourage you to do a base generic class.

```
internal class History<T>
{
 List<T> m_Samples = new();
 public History(int initialCapacity):
 public void DiscardOldSamples(int sampleHistorySize);
 public void Add(T sample);
 public void Clear();
}

class BottleneckHistory : History<PerformanceBottleneck> and FrameTimeSampleHistory : History<FrameTimeSample>

Reply:
Yes, I had the same thought when making the same changes to both files, but decided to not tackle the issue right now :) I do agree this would make sense though. I'll make a note to come back to this later.